src/gdrapi.c

```diff
@@ -537,59 +537,70 @@ static inline int ptr_is_aligned(const void *ptr, unsigned powof2)
 
 static int gdr_copy_to_mapping_internal(void *map_d_ptr, const void *h_ptr, size_t size, int wc_mapping)
 {
-    if (first_time) {
-        gdr_init_cpu_flags();
-    }
-
     do {
+        // For very small sizes and aligned pointers, we use simple store.
+        if (unlikely(size == 0)) {
+            goto out;
+        } else if (size == 1) {
+            WRITE_ONCE(*(uint8_t *)map_d_ptr, *(uint8_t *)h_ptr);
+            goto do_fence;
+        } else if (size == 2 && ptr_is_aligned(map_d_ptr, 2) && ptr_is_aligned(h_ptr, 2)) {
```

Comment:
can you use sizeof(uint16_t) instead of 2 ?

Reply:
I think that you can relaxed the alignment check on the source, as that will always be a pointer to normal memory, most probably on the stack or even a register if LTO inlines this API.

Reply:
Fixed.